src/Symfony/Component/Filesystem/Filesystem.php

```diff
@@ -523,6 +523,7 @@ public function dumpFile($filename, $content)
         }
 
         $tmpFile = $this->tempnam($dir, basename($filename));
+        $this->chmod($tmpFile, 0666, umask());
```

Comment:
Apply the mode change after writing to the file the umask may remove the write permission.